### ISSUE TITLE
[REQUEST] improved compatibility with local (file://) pages and protocol-relative URLs

### ISSUE THREAD

**Issue by jokeyrhyme** [requirejs/requirejs#1058]
**TL;DR**: please detect URLs that start with `//` and automatically prepend `http:` when `location.protocol === 'file:'`

I've been duck-punching this in myself for a while now, and it occurs to me that someone else might like this functionality.

I have a collection of web apps, some of these run via HTTP, and others via HTTPS. I load shared libraries for these web apps with protocol-relative URLs, e.g. `//cdn.domain.com/path/to/resource`.

Further, for some of my unit tests, it is preferable if I can just load them as local files, instead of needing to temporarily host them within a static web server. I can definitely do the latter, but it would be far faster and more convenient to just open the files directly, e.g. `file:///path/to/unit/test.html`.

Chrome, and probably other browsers, don't know how to handle requests for protocol-relative URLs when the current document is a local file.

This request would allow me the convenience of testing with local files without needing to change my protocol-relative URLs, and without having to temporarily host the tess (statically or otherwise).

I'm completely in awe of you and your work. Thanks!
